### ISSUE TITLE
Error in compilation with Nuitka

### ISSUE THREAD

**Issue by jbdo99** [pokepetter/ursina#20]
Hi !

Very interested in the ursina project, I wanted to test a little puzzle coding to test the game engine.
However, once finished, I tried to compile it with nuitka simply using: 
`python -m nuitka --standalone --mingw64 --show-progress --show-scons  --follow-imports khun-pan.py`
(the code : [here](https://gist.github.com/jbdo99/618440279339f9fc2e8a9ea1e6050c17))

I ended up with this error, which seems to be related to panda3d.

```
Warning: unable to auto-locate config files in directory named by "<auto>etc".
package_folder: D:\Ursina\Test_Project\khun_pan\khun-pan.dist\ursina
asset_folder: D:\Ursina\Test_Project\khun_pan\khun-pan.dist
blender_paths:
{}
psd-tools not installed
missing model: quad
Known pipe types:
(all display modules loaded.)
Traceback (most recent call last):
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\khun-pan.py", line 150, in <module>
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\ursina\main.py", line 9, in __init__
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\direct\showbase\ShowBase.py", line 289, in __init__
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\direct\showbase\ShowBase.py", line 959, in openDefaultWindow
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\direct\showbase\ShowBase.py", line 995, in openMainWindow
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\direct\showbase\ShowBase.py", line 707, in openWindow
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\direct\showbase\ShowBase.py", line 687, in <lambda>
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\direct\showbase\ShowBase.py", line 759, in _doOpenWindow
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\direct\showbase\ShowBase.py", line 597, in makeDefaultPipe
  File "D:\Ursina\Test_Project\khun_pan\khun-pan.dist\direct\directnotify\Notifier.py", line 130, in error
Exception: No graphics pipe is available!
Your Config.prc file must name at least one valid panda display
library via load-display or aux-display.

```

Any idea how to fix it?

Thanks a lot.

**Comment by clayboone** [comment]
I don't know what Nuitka is, but based on that output, it looks like it uses the `direct` package. Ursina also uses this package and it looks like you've made your module _inside_ of the Ursina root folder. This is a good way to ensure import problems.

Can you test this by moving your script to some other location? Or try replicating it without using Nuitka?

These might be obvious or impossible questions, but I'm curious to know if this is a challenge for Ursina or Nuitka, or if it's just a local configuration issue.

**Comment by jbdo99** [comment]
Nuitka is one of the best compilers for python, and I didn't see that there was a script to build an application in ursina. My script is not in the ursina files. This problem seems to be more related to Ursina because I also tested with pyinstaller, and I end up with the same error. I'll try to compile ursina alone.

P.S: The idea to use nuitka came from @pokepetter  who had made a [tweet](https://twitter.com/ursinaengine/status/1222489946268803072), and who also answered me.

Thanks for the help :)

**Comment by clayboone** [comment]
Well I can at least confirm this: I've just tried with PyInstaller 3.6 and got the same result as you.

**Comment by jbdo99** [comment]
Ok it seems that the problem is from panda3d, I tested the compilation of a panda3d sample with the same parameters, and I get the same error.

**Comment by clayboone** [comment]
Looks like we had the same idea. I just came to report that failure as well. Additionally, when I try out the `python -m ursina.build`, I get some unicode error. That should probably be another ticket, but I think we can close this one for now.

This is a Panda3d limitation and, based on their documentation, not likely to be fixed.

**Comment by jbdo99** [comment]
I'm going to close this ticket because in the end I found the solution to my problem (making the games developed with ursina distributable)
I'm putting here the latest research and tests I've done, if anyone wants to take it over later:

After several tests I've managed to get nuitka to work with the basic example of panda3d (that horrible panda that spins)
For this I had to do several manual manipulations after the nuitka execution:
The first error comes from the lack of panda3d configuration files in the compilation. So I moved the /etc/ folder with its two configuration files into the dist directory. ( panda3d location $PYTHON_PATH$\Lib\site-packages\panda3d )
Then, the problem is that it doesn't find the display libs (libpandagl.dll), so I added these lines in the configuration :
```
plugin-path   $MAIN_DIR
plugin-path   $THIS_PRC_DIR/..
plugin-path   $THIS_PRC_DIR/../lib
```
I then moved libpandagl to the dist directory.
I then got the following error : 
```
:display(warning): Unable to load libpandagl.so: Module not found
```
To solve this I moved all the files (.dll and .pyd) in the dist directory. And it all worked.
I couldn't reproduce the result with ursina, strange bug on missing variables in the code.

Thanks for the help!
